xbmc/cores/VideoPlayer/DVDCodecs/Overlay/DVDOverlayCodecCCText.cpp

```diff
@@ -0,0 +1,150 @@
+/*
+ *  Copyright (C) 2005-2021 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "DVDOverlayCodecCCText.h"
+
+#include "DVDCodecs/DVDCodecs.h"
+#include "DVDOverlayText.h"
+#include "DVDStreamInfo.h"
+#include "cores/VideoPlayer/DVDSubtitles/DVDSubtitleTagSami.h"
+#include "cores/VideoPlayer/Interface/DemuxPacket.h"
+#include "utils/log.h"
+
+#include "system.h"
+
+namespace
+{
+// Muxed Closed Caption subtitles don't have the stop PTS value,
+// the stop value can be taken from: the start PTS of the next subtitle,
+// or, the start PTS of the next subtitle without text -if any-,
+// otherwise we fallback to 20 secs duration (long duration so as not
+// to cause side effects on texts that expand like karaoke, but slowly)
+constexpr double DEFAULT_DURATION = 20.0 * (double)DVD_TIME_BASE;
+} // namespace
+
+CDVDOverlayCodecCCText::CDVDOverlayCodecCCText() : CDVDOverlayCodec("CC Text Subtitle Decoder")
+{
+  m_pOverlay = nullptr;
+  m_prevSubId = NO_SUBTITLE_ID;
+  m_prevPTSStart = 0.0;
+  m_prevText.clear();
+  m_changePrevStopTime = false;
+}
+
+CDVDOverlayCodecCCText::~CDVDOverlayCodecCCText()
+{
+  Dispose();
+}
+
+bool CDVDOverlayCodecCCText::Open(CDVDStreamInfo& hints, CDVDCodecOptions& options)
+{
+  Dispose();
+
+  return Initialize();
+}
+
+void CDVDOverlayCodecCCText::Dispose()
+{
+  if (m_pOverlay)
```

Comment:
`SAFE_RELEASE` macro already checks for object not NULL so `if (m_pOverlay)` condition can be removed

Reply:
Please don't use `SAFE_RELEASE` macro, I've been trying to get rid of it for a while now.

Reply:
can go well if replace all SAFE_RELEASE with:
```
  if (m_pOverlay)
  {
    m_pOverlay->Release();
    m_pOverlay = nullptr;
  }
```
?

Reply:
Then:

```c++
  if (m_pOverlay)
  {
    m_pOverlay->Release();
    m_pOverlay = nullptr;
  } 
```

Reply:
done